deploy_pulp3/pulp3/tasks/main.yml

```diff
@@ -0,0 +1,85 @@
+---
+# tasks file for pulp3
+- name: Create user {{ pulp_user }}
+  user:
+    name: "{{ pulp_user }}"
+    comment: "Pulp user"
+
+- name: Get user's home directory
+  shell: 'getent passwd "{{ pulp_user }}" | cut -d: -f6'
+  check_mode: false
+  changed_when: false
+  register: result
+
+- name: Set variable referencing user's home directory
+  set_fact:
+    pulp_user_home: "{{ result.stdout }}"
+
+- name: Make user's home directory world-readable
+  file:
+    path: "{{ pulp_user_home }}"
+    state: directory
+    mode: 0755
+
+- name: Create ~/.bashrc.d/
+  file:
+    path: "{{ pulp_user_home }}/.bashrc.d/"
+    state: directory
+  become_user: "{{ pulp_user }}"
+
+- name: Create directory /var/lib/pulp
+  file:
+    path: /var/lib/pulp
+    state: directory
+    owner: "{{ pulp_user }}"
+    group: "{{ pulp_user }}"
+    mode: 0775
+
+- name: Install pulpcore
+  pip:
+    name: pulpcore
+    state: latest
+    virtualenv: "{{ pulp_user_home }}/pulpvenv"
+    virtualenv_python: python3
+  become_user: "{{ pulp_user }}"
+
+- name: Add Django and Pulp supplemental bashrc
+  copy:
+    src: files/supplem.bashrc
+    dest: "{{pulp_user_home}}/.bashrc.d/suppelm.bashrc"
+
+- name: Source  ~/.bashrc.d/
+  blockinfile:
+    path: "{{ pulp_user_home }}/.bashrc"
+    block: |
+      if [ -d ~/.bashrc.d ]; then
+        for file in ~/.bashrc.d/*; do
+          . "$file"
+        done
+      fi
+    marker: "# {mark} Ansible managed block: source bashrc.d/"
+    create: yes
+  become_user: "{{ pulp_user }}"
+
+- name: Create /etc/pulp dir
+  file:
+    path: /etc/pulp
+    state: directory
+
+- name: Install server.yaml Config file
+  copy:
+    remote_src: true
+    src: "{{pulp_venv}}/lib/python3.5/site-packages/pulpcore/etc/pulp/server.yaml"
+    dest: "/etc/pulp/server.yaml"
+
+- name: Generate random secret for Django
+  shell: cat /dev/urandom | tr -dc 'a-z0-9!@#$%^&*(\-_=+)' | head -c 50
+  register: random_secret
+
+- name: Add development server configuration in server.yaml
+  blockinfile:
+    path: "/etc/pulp/server.yaml"
+    block: |
+      # Pulp Development Configuration
+      DEBUG: True
+      SECRET_KEY: {{ random_secret.stdout | to_nice_yaml }}
```

Comment:
It might be better to use two `lineinfile` tasks instead of this `blockinfile` task. Again, this is to promote idempotency. Given that `random_secret` will vary every time this play runs, this block will be updated every time this task runs, which is not Ansible-ish.

Reply:
We should not be setting debug: True and generating a secret key for a non dev install of pulp. This task and the previous task should be removed.